python_modules/dagster/dagster/_core/definitions/decorators/repository_decorator.py

```diff
@@ -164,16 +156,26 @@ def __call__(
                     f"Got {bad_definitions_str}."
                 )
 
-            repository_data = (
-                None
-                if defer_repository_data
-                else CachingRepositoryData.from_list(
+            if defer_repository_data:
```

Comment:
Is this only true is there is a cacheable assets definition? If so, consider renaming to something more literal: `has_cacheable_assets_def` or something? That way the person doesn't have to look up to see what the logic is.

Reply:
>Is this only true is there is a cacheable assets definition?

Yes-- good idea, renamed